redfish-core/lib/power_supply_metrics.hpp

```diff
@@ -187,110 +189,199 @@ inline void
+            {
+                if (service.empty())
+                {
+                    BMCWEB_LOG_DEBUG << "Error getting D-Bus object!";
+                    messages::internalError(aResp->res);
+                    return;
+                }
+
+                for (const auto& interface : interfaceNames)
+                {
+                    if (interface == averageInterface)
+                    {
+                        if (serviceName.empty())
+                        {
+                            serviceName = service;
+                        }
+                        if (averagePath.empty())
+                        {
+                            averagePath = objectPath;
+                        }
+                    }
+                    else if (interface == maximumInterface)
+                    {
+                        if (serviceName.empty())
+                        {
+                            serviceName = service;
+                        }
+                        if (maximumPath.empty())
+                        {
+                            maximumPath = objectPath;
+                        }
+                    }
+                }
+                BMCWEB_LOG_DEBUG << "serviceName: " << serviceName;
+                BMCWEB_LOG_DEBUG << "averagePath: " << averagePath;
+                BMCWEB_LOG_DEBUG << "maximumPath: " << maximumPath;
+            }
+
+            if (objectPath2.empty())
+            {
+                BMCWEB_LOG_DEBUG << "Get power supply date/average/maximum "
+                                    "input power values";
+                getAverageMaximumValues(aResp, serviceName, averagePath,
+                                        maximumPath);
+            }
+            else
+            {
+                getServicePathValues(aResp, objectPath2, "", serviceName,
+                                     averagePath, maximumPath);
+            }
+        };
+
+    crow::connections::systemBus->async_method_call(
+        getServiceAndPathHandler, "xyz.openbmc_project.ObjectMapper",
+        "/xyz/openbmc_project/object_mapper",
+        "xyz.openbmc_project.ObjectMapper", "GetObject", objectPath,
+        interfaces);
+}
+
 /**
- * @brief Get power supply average, maximum and date values given chassis and
- * power supply IDs.
+ * @brief Get power supply average, maximum and date values given chassis
+ * and power supply IDs.
  *
  * @param[in] aResp - Shared pointer for asynchronous calls.
- * @param[in] chassisID - Chassis to which the values are associated.
- * @param[in] powerSupplyID - Power supply to which the values are associated.
+ * @param[in] inputHistoryItem - Array of object paths for input history.
  *
  * @return None.
  */
 inline void getValues(const std::shared_ptr<bmcweb::AsyncResp>& aResp,
-                      const std::string& chassisID,
-                      const std::string& powerSupplyID)
+                      const std::vector<std::string>& inputHistoryItem)
```

Comment:
Nit: Should this be plural inputHistoryItems?

Reply:
I have seen people do this both ways. I initially called it inputHistoryItems, but then wondered if it made more sense when it was inputHistoryItem[0], etc. So, I went with singular.

Reply:
I think Items would have been better but shrug